### DISCUSSION TITLE
feature idea: compilation-time llm-integration to explain code blocks

### DISCUSSION THREAD

**Discussion by EqualAPriori** [quarto-dev/quarto-cli#10002] [Feature Requests]
### Description

Hi all, wanted to get feedback on a feature that I'd be interested in helping out with.

The idea is to add a compilation-time flag to code blocks like thus:

`#| llm-explain: true`

Upon compilation time, instead of printing the code block (as would happen with `#| echo: true`), to instead feed the code block to a llm for interpretation.

Other details:
- include API key + llm service of choice in the pre-amble config section of the notebook.
- harder is to have options for interpreting the outputs as well.

Apologies if this is under consideration somewhere, I may have missed it in my searches.

Do people think this would be interesting? This would be my first time contributing to the codebase, what would be the best way to go about that?

**Comment by mcanouil** [comment]
That would be an extension, see https://quarto.org/docs/extensions/creating.html

**Reply by mcanouil** [reply]
Additionally, it's unlikely Quarto will take such dependencies as its core, so the extension is the best way to provide such feature.

**Comment by cscheid** [comment]
Thanks for the suggestion, but let me be clear: we're not going to do that in quarto-cli - we have no interest in adding such a deep dependency to LLMs into a tool that needs to work in offline settings. I would never say never, but I will say "not while I'm the main contributor"
